src/contradictory_claims/data/process_claims.py

```diff
@@ -0,0 +1,162 @@
+
+    :param input_data: pandas dataframe with publication text
+    :retunr: Dataframe with section text tokenized to sentences
+    """
+    # Empty dictonary to store the tokenized text
+    text_dict = {}
+    # Dictionary iterator
+    k = 0
+
+    # Loop through the sections and tokenize text to sentences
+    for i, text in enumerate(input_data.text):
+        for sent in sent_tokenize(text):
+            text_dict[k] = {'cord_uid': input_data.cord_uid[i],
+                            'section': input_data.section[i],
+                            'text': input_data.text[i],
+                            'drug_terms_used': input_data.drug_terms_used[i],
+                            'claims': sent}
+            k = k + 1
+
+    return pd.DataFrame.from_dict(text_dict, "index")
+
+
+def pair_similar_claims(claims_data: pd.DataFrame, nlp):
+    """
+    Pair similar claims.
+
+    :param claims_data: pandas dataframe with cord 19 claims
+    :param nlp: Scispacy nlp object
+    :return: Dataframe of paired claims
+    """
+    # Extract list of drug terms present across all claims
+    # Note: 'drug_terms_used' consists of drug terms present in the section in which the claim appears
+    drug_terms = []
+    for drugs in claims_data.drug_terms_used:
+        drug_terms = drug_terms + str(drugs).split(',')
+    drug_terms = list(set(drug_terms))
+    drug_terms.append('acei/arb')
+
+    # Filter to claims that contain drug terms
+    sentences_to_keep = [any(True for d in drug_terms if d in c) for c in claims_data.claims]
+    claims_data = claims_data[sentences_to_keep].reset_index(drop=True)
+    # Add a new column for storing the drug terms present in each claim
+    claims_data['drug_terms_mention'] = [[d for d in drug_terms if d in c] for c in claims_data.claims]
+
+    # Pair all claims
+    paper_pairs = list(combinations(claims_data.index, 2))
+    paper_pairs_filt = []
+    # Filter to claim pairs that come from different papers and have at least 1 drug term in common
+    for i, j in paper_pairs:
+        if claims_data.cord_uid[i] != claims_data.cord_uid[j]:
+            if any(d1 in claims_data.drug_terms_mention[i] for d1 in claims_data.drug_terms_mention[j]):
+                paper_pairs_filt.append((i, j))
+
+    # Calculate scispacy vector for each claim
+    claims_data['w2vVector'] = [nlp(c).vector.reshape(1, -1) for c in claims_data.claims]
+
+    # Empty dictonary to store the similar claim pairs
+    claim_pairs_dict = {}
+    # Dictionary iterator
+    k = 0
+
+    # Initialize just-in-time compiler for efficient parallel processing
+    jit(nopython=True, parallel=True)
+
+    # For each pair of claims, calculate cosine similarity between the respective scispacy vectors
+    # and keep only those pairs with at least 50% similarity
+    for i, j in paper_pairs_filt:
+        cos_sim = cosine_similarity(claims_data.w2vVector[i], claims_data.w2vVector[j])[0][0]
+        if cos_sim >= 0.5:
```

Comment:
Okay, here's where we'll have to innovate. For starters, have you played with changing this threshold? Beyond that we can try some form of polarity detection/context matching (maybe even with something simple like TF-IDF to find common, important words like "pneumonia")

Reply:
Yeah agree, looking at polarity/context matching is on my list of things for this week.
I did look at a random sample of the results  for scores < 50% and they seemed okay to ignore. For cases with higher scores, I don't remember being able to make a definite call so I kept them.